### DISCUSSION TITLE
What would cause the path to be empty?

### DISCUSSION THREAD

**Discussion by 6ther** [isaac-mason/recast-navigation-js#279] [General]
`const positions = [
        173.07352, 0, -131.28241,
        173.07352, 0, 50.717583, 
        -88.92649, 0, 50.717583, 
        -88.92649, 0, -84.2824, 
        -45.926494, 0, -84.2824, 
        -45.926494, 0, -9.2824164, 
        83.07351, 0, -9.2824164,
        83.07351, 0, -131.28241, 
        173.07352, 0, -131.28241, 
    ];
      
    const indices = [
        8, 1, 2, 
        3, 4, 5, 
        7, 8, 2, 
        3, 5, 6, 
        6, 7, 2, 
        2, 3, 6
    ];
      
    const navMeshConfig = {
        borderSize:0,
        ch:0.2,
        cs:0.2,
        detailSampleDist:6,
        detailSampleMaxError:1,
        maxEdgeLen:12,
        maxSimplicationError:1.3,
        maxVertsPerPoly:6,
        mergeRegionArea:20,
        minRegionArea:8,
        titleSize:0,
        walkableClimb:2,
        walkableHeight:2,
        walkableRedius: 1,
        walkableSlopeAngle:60
    };
      
    const { success, navMesh } = generateSoloNavMesh(
        positions,
        indices,
        navMeshConfig
    );
  const navMeshQuery = new NavMeshQuery({ navMesh });
 const path = navMeshQuery.computePath(
        { x: -88.92649, y: 0, z: -84.2824}, // start position
        { x: 173.07352, y: 0, z: -131.28241 } // end position
    );`
navMeshQuery.computePath return the path is empty.
And the Navmesh is right as shown in the figure.
![image](https://github.com/isaac-mason/recast-navigation-js/assets/46700515/db74ed68-26d7-4f98-981e-72362227fed6)

**Comment by isaac-mason** [comment]
Hi @6ther!

Something you could try is passing `computePath` the results from `navMeshQuery.getClosestPoint`, for example:

```ts
const path = navMeshQuery.computePath(
  navMeshQuery.getClosestPoint({ x: -88.92649, y: 0, z: -84.2824}), // start position
  navMeshQuery.getClosestPoint({ x: 173.07352, y: 0, z: -131.28241 }) // end position
);
```

Also, do you mind sharing the file you uploaded to navmesh.isaacmason.com so I can test with that as well?

The geometry I've created using the positions and indices provided looks different: 

<img width="879" alt="image" src="https://github.com/isaac-mason/recast-navigation-js/assets/67411435/b34a28c5-2b33-49ce-848d-3a3c57aec341">

Let me know if I'm doing something silly.

**Comment by 6ther** [comment]
[planeXunlu.txt](https://github.com/isaac-mason/recast-navigation-js/files/14503444/planeXunlu.txt)
`  
 const path = navMeshQuery.computePath(
        navMeshQuery.getClosestPoint({ x: 88.92649, y: 0, z: -84.2824}), // start position
        navMeshQuery.getClosestPoint({ x: -173.07352, y: 0, z: -131.28241 }) // end position
    );
`
{ x: -88.92649, y: 0, z: -84.2824}，{ x: 173.07352, y: 0, z: -131.28241 } these points is not in NavMesh, so the path is empty.
But I try another points:{ x: 88.92649, y: 0, z: -84.2824},{ x: -173.07352, y: 0, z: -131.28241 },
the  path is 
[
  (88.72650146484375, 0.20000000298023224, -84.08241271972656),
  (26.5264892578125, 0.20000000298023224, -56.882408142089844),
  (3.1264801025390625, 0.20000000298023224, -46.682411193847656),
  (1.7264862060546875, 0.20000000298023224, -46.08240509033203),
  (-1.2735137939453125, 0.20000000298023224, -44.882408142089844),
  (-172.87351989746094, 0.20000000298023224, -131.08241271972656)
],
shown in this image:
![image](https://github.com/isaac-mason/recast-navigation-js/assets/46700515/8ea920fc-5598-40c5-a7ab-ae2f83fb8f7e)
the path is not completely on the NavMesh. I do not know why?

**Comment by isaac-mason** [comment]
Hey @6ther, could you could share a [codesandbox](https://codesandbox.io/) (or similar) reproduction? I'm having some trouble recreating your environment so I can debug.

Using the obj model, the navmesh config provided, and the given start and end points, I get a different result:

<img width="1352" alt="image" src="https://github.com/isaac-mason/recast-navigation-js/assets/67411435/83afdd67-e96c-47f8-abf4-b2ae8d5790e2">

**Comment by 6ther** [comment]
My code like this:
`
import { init } from "recast-navigation";
import { generateSoloNavMesh } from "recast-navigation/generators";
import { NavMeshQuery } from 'recast-navigation';
import { Crowd } from 'recast-navigation';

export async function getNavMesh()
{
    await init();
    const positions = [
        -173.07352, 0, -131.28241,
        -173.07352, 0, 50.717583, 
        88.92649, 0, 50.717583, 
        88.92649, 0, -84.2824, 
        45.926494, 0, -84.2824, 
        45.926494, 0, -9.2824164, 
        -83.07351, 0, -9.2824164,
        -83.07351, 0, -131.28241, 
        -173.07352, 0, -131.28241, 
    ];
      
    const indices = [
        8, 1, 2, 
        3, 4, 5, 
        7, 8, 2, 
        3, 5, 6, 
        6, 7, 2, 
        2, 3, 6
    ];
      
    const navMeshConfig = {
        borderSize:0,
        ch:0.2,
        cs:0.2,
        detailSampleDist:6,
        detailSampleMaxError:1,
        maxEdgeLen:12,
        maxSimplicationError:1.3,
        maxVertsPerPoly:6,
        mergeRegionArea:20,
        minRegionArea:8,
        titleSize:0,
        walkableClimb:2,
        walkableHeight:2,
        walkableRedius: 1,
        walkableSlopeAngle:60
    };
      
    const { success, navMesh } = generateSoloNavMesh(
        positions,
        indices,
        navMeshConfig
    );

    const navMeshQuery = new NavMeshQuery({ navMesh });


    const path = navMeshQuery.computePath(
        navMeshQuery.getClosestPoint({ x: 88.92649, y: 0, z: -84.2824}), // start position
        navMeshQuery.getClosestPoint({ x: -173.07352, y: 0, z: -131.28241 }) // end position
    );

    const stringArray: string[] = [];
    path.forEach(p => stringArray.push(`(${p.x}, ${p.y}, ${p.z})`))
    console.log(stringArray);
}
`

**Comment by isaac-mason** [comment]
Hi @6ther,

I think this is an input issue. There is some difference between the obj model and the positions and indices in the code snippets.

You can visualise the resulting navmesh generated by the above snippet using `NavMeshHelper` from `recast-navigation/three`:

<img width="1274" alt="image" src="https://github.com/isaac-mason/recast-navigation-js/assets/67411435/493ecb3a-63d8-4324-8ce1-59840b368035">

Creating a buffer geometry shows a geometry that matches the navmesh:

<img width="1095" alt="image" src="https://github.com/isaac-mason/recast-navigation-js/assets/67411435/db7dcb94-ce2c-4148-b059-a826bac972f6">
